### DISCUSSION TITLE
ffmpeg filters - can they improve frigate detection and if so, how to use them?

### DISCUSSION THREAD

**Discussion by OttoL2** [blakeblackshear/frigate#9746] [Ask A Question]
Hi all, potentially stupid question but maybe I can save myself a lot of time if there's an easy answer. I use frigate with a coral tpu and 2 TPLink TAPOC310 cameras. I use it from within Home Assistant. The Tapo cameras, whilst cheap and cheerful an actually doing the job pretty well, have no possibilities to modify brightness, gamma, contrast. Due to the place I have to mount it, this results in one camera giving a very dark image (well the biggest part of the view). Switching to night/IR mode (that IS possible on the cameras) is not much help as the IR leds do not reach far enough. So, my question - could I pass some parameters to ffmpeg to brighten the image before Frigate evaluates it , and would that have any effect on detection accuracy/rate? In other words, can a filter in ffmpeg lead to better detection?

I apologise if this demonstrates a gross misunderstanding of the workings of an otherwise very fine program!
Maarten

**Comment by NickM-27** [comment]
in general these types of filters increase cpu usage and require re-encoding the stream if you want to use them for recording. It is possible to do more efficiently if only changing for object detection but that is still a bit complicated and will increase cpu usage

It can improve object detection results but I imagine results will vary

**Reply by OttoL2** [reply]
Thanks for the quick answer Nick. If I still want to try this (I have HA and Frigate running on a dedicated I5 with plenty memory so probably some cycles spare) would there be an example somewhere that you know of, of ffmpeg parameters I could include into the frigate.yml file to achieve this?
Thanks, 
Maarten

**Reply by NickM-27** [reply]
it depends if you want it to affect recordings too or just detect

**Reply by OttoL2** [reply]
Would be for detection only...

**Reply by NickM-27** [reply]
you would add to detect output_args then. https://video.stackexchange.com/questions/20962/ffmpeg-color-correction-gamma-brightness-and-saturation
